zigpy/ota/provider.py

```diff
@@ -535,3 +536,85 @@ async def refresh_firmware_list(self) -> None:
                     self._cache[img.key] = img
 
         self.update_expiration()
+
+
+@attr.s
+class INOVELLIImage:
+    manufacturer_id = attr.ib()
+    image_type = attr.ib()
```

Comment:
The purpose of the `INOVELLIImage` object is to keep around a reference to the original image, so that it can be downloaded on-demand. It should contain all of relevant information about an image, returned by your JSON API.

Here's what the JSON response from your server is currently returning for a device:

```JSON
{
	"VMDS00I00": [{
		"version": "00000005",
		"channel": "beta",
		"firmware": "https://files.inovelli.com/firmware/VZM31-SN/Beta/00000005/00000005.ota"
	}]
}
```

Your `INOVELLIImage` class would therefore have the following attributes:

 - `model` (from the dictionary key)
 - `version`
 - `channel` (if useful)
 - `url` (called `firmware` in the JSON)

Since there is no other information encoded in the JSON, you can't include `image_type` or `image_size` (since those are not known).